src/Symfony/Component/Cache/Adapter/CouchbaseBucketAdapter.php

```diff
@@ -0,0 +1,226 @@
+
+                $username = $matchers['username'] ?: $username;
+                $password = $matchers['password'] ?: $password;
+                $newServers += explode(',', $matchers['cluster']);
+            }
+
+            $servers = $newServers ?? $servers;
+            $connectionString = $matchers['protocol'].'://'.implode(',', $servers);
+
+            $client = new \CouchbaseCluster($connectionString);
+            $client->authenticateAs($username, $password);
+
+            return $client;
+        } finally {
+            restore_error_handler();
+        }
+    }
+
+    /**
+     * @return bool
+     */
+    public static function isSupported(): bool
+    {
+        return \extension_loaded('couchbase') && version_compare(phpversion('couchbase'), '2.6.0', '>=');
+    }
+
+    /**
+     * @param \CouchbaseCluster        $client
+     * @param string                   $bucket
+     * @param string                   $namespace
+     * @param int                      $defaultLifetime
+     * @param array                    $options
+     * @param MarshallerInterface|null $marshaller
+     *
+     * @throws CacheException
+     */
+    private function init(
+        \CouchbaseCluster $client,
+        string $bucket,
+        string $namespace,
+        int $defaultLifetime,
+        array $options = [],
+        ?MarshallerInterface $marshaller = null
+    ): void {
+        if (!static::isSupported()) {
+            throw new CacheException('Couchbase >= 2.6.0 is required');
+        }
+
+        $this->maxIdLength = static::MAX_KEY_LENGTH;
+
+        $this->bucket = $client->openBucket($bucket);
+
+        $this->bucket->operationTimeout = $options['operationTimeout'] ?? $this->bucket->operationTimeout;
+        $this->bucket->configTimeout = $options['configTimeout'] ?? $this->bucket->configTimeout;
+        $this->bucket->configNodeTimeout = $options['configNodeTimeout'] ?? $this->bucket->configNodeTimeout;
+
+        parent::__construct($namespace, $defaultLifetime);
+        $this->enableVersioning();
+        $this->marshaller = $marshaller ?? new DefaultMarshaller();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFetch(array $ids)
+    {
+        $resultsCouchbase = $this->bucket->get($ids);
+
+        $results = [];
+        foreach ($resultsCouchbase as $key => $value) {
+            if (null !== $value->error) {
+                continue;
+            }
+            $results[$key] = $this->marshaller->unmarshall($value->value);
+        }
+
+        return $results;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doHave($id): bool
+    {
+        return false !== $this->bucket->get($id);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doClear($namespace): bool
+    {
+        if ('' === $namespace) {
+            $this->bucket->manager()->flush();
+
+            return true;
+        }
+
+        return false;
```

Comment:
no way to flush by prefix?

Reply:
Couchbase can not search for something without having an index on that data, that means that the bucket destined to cache would have to be indexed and that would cause an overload and a slowdown of the cache.
I can flush all the contents of the bucket, if the bucket allows flush, but never over a specific namespace.